test/Microsoft.Health.Fhir.Shared.Tests.Integration/Features/Operations/Reindex/ReindexJobTests.cs

```diff
@@ -355,35 +438,34 @@ private async Task<CreateReindexResponse> SetUpForReindexing()
             return response;
         }
 
-        private void MockSearchIndexExtraction(string sampleName1, string sampleName2, SearchParameter searchParam)
+        private void MockSearchIndexExtraction(IEnumerable<(string id, ISearchValue searchValue)> searchValues, SearchParameter searchParam)
         {
             SearchParameterInfo searchParamInfo = searchParam.ToInfo();
 
-            var searchIndexValues1 = new List<SearchIndexEntry>();
-            searchIndexValues1.Add(new SearchIndexEntry(searchParamInfo, new StringSearchValue(sampleName1)));
-            _searchIndexer.Extract(Arg.Is<ResourceElement>(r => r.Id.Equals(sampleName1))).Returns(searchIndexValues1);
-
-            var searchIndexValues2 = new List<SearchIndexEntry>();
-            searchIndexValues2.Add(new SearchIndexEntry(searchParamInfo, new StringSearchValue(sampleName2)));
-            _searchIndexer.Extract(Arg.Is<ResourceElement>(r => r.Id.Equals(sampleName2))).Returns(searchIndexValues2);
+            foreach ((string id, ISearchValue searchValue) in searchValues)
+            {
+                var searchIndexValues = new List<SearchIndexEntry>();
+                searchIndexValues.Add(new SearchIndexEntry(searchParamInfo, searchValue));
+                _searchIndexer.Extract(Arg.Is<ResourceElement>(r => r.Id.Equals(id))).Returns(searchIndexValues);
+            }
         }
 
-        private async Task<SearchParameter> CreateSearchParam(string searchParamName, string searchParamCode)
+        private async Task<SearchParameter> CreateSearchParam(string searchParamName, SearchParamType searchParamType, ResourceType baseType, string expression, string searchParamCode)
         {
-            var searchParam = new SearchParameter()
+            var searchParam = new SearchParameter
             {
-                Url = $"http://hl7.org/fhir/SearchParameter/Patient-{searchParamName}",
-                Type = SearchParamType.String,
-                Base = new List<ResourceType?>() { ResourceType.Patient },
-                Expression = "Patient.name",
+                Url = $"http://hl7.org/fhir/SearchParameter/{baseType.ToString()}-{searchParamName}",
```

Comment:
>ToString() [](start = 70, length = 10)

isn't ToString() applies automatically to object if you pass it this way?

Reply:
Ah yes you're right. Fixed in 7555909.